### ISSUE TITLE
Residual wave after pulse or ending continuous

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/wave-interference#251]
**Test device:**
iPad and Dell
**Operating System:**
iOS 12 and Win 10
**Browser:**
safari and chrome
**Problem description:**
For https://github.com/phetsims/QA/issues/238
When a single wave is produced, or a final wave front is made, there appear to be clear, smaller wave fronts moving behind it.
**Steps to reproduce:**
1. Go to the wave screen 
2. Go to the light scene and select a bright, easy to see color
3. Select pulse
4. Send out a single wave

**Screenshots:**
![residualwavesinblue](https://user-images.githubusercontent.com/41024075/49974797-320f2d00-fef8-11e8-9605-9b449649f094.PNG)
https://drive.google.com/file/d/11qfg32PuwV9JeFQR9cxv43WTXSi7VPFM/view?usp=sharing

Troubleshooting information (do not edit):

<details>

Name: ‪Wave Interference‬
URL: https://phet-dev.colorado.edu/html/wave-interference/1.0.0-dev.51/phet/wave-interference_en_phet.html
Version: 1.0.0-dev.51 2018-12-12 03:47:24 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36
Language: en-US
Window: 1536x732
Pixel Ratio: 2.5/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"928741cf","branch":"master"},"axon":{"sha":"73e761ec","branch":"master"},"brand":{"sha":"1fd6682e","branch":"master"},"chipper":{"sha":"61bc739b","branch":"master"},"dot":{"sha":"b45eafe8","branch":"master"},"griddle":{"sha":"914c9215","branch":"master"},"joist":{"sha":"5a013a46","branch":"master"},"kite":{"sha":"f3e96e5b","branch":"master"},"phet-core":{"sha":"86dec8cc","branch":"master"},"phet-io":{"sha":"a267fdb1","branch":"master"},"phet-io-wrapper-classroom-activity":{"sha":"5dea7f54","branch":"master"},"phet-io-wrapper-hookes-law-energy":{"sha":"b42eaef7","branch":"master"},"phet-io-wrapper-lab-book":{"sha":"b884e9c1","branch":"master"},"phet-io-wrappers":{"sha":"d5b42848","branch":"master"},"phetcommon":{"sha":"869b2561","branch":"master"},"query-string-machine":{"sha":"e4f9e8e8","branch":"master"},"scenery":{"sha":"a18ce7f1","branch":"master"},"scenery-phet":{"sha":"56df5b6d","branch":"master"},"sherpa":{"sha":"a1f5e867","branch":"master"},"sun":{"sha":"38ed71af","branch":"master"},"tambo":{"sha":"746fbd32","branch":"master"},"tandem":{"sha":"c7a56e2e","branch":"master"},"twixt":{"sha":"e38c70ed","branch":"master"},"wave-interference":{"sha":"bfce593d","branch":"master"}}

</details>

**Comment by samreid** [comment]
I believe the cause of the extra ripples is that the wave is being suddenly set to 0.  It seems our options are to live with this behavior or use something like exponential decay to slowly restore the oscillator value back to zero. I'm not sure what the period would be for this new decay phase, nor do I know whether it would introduce undesirable artifacts or behavior.  @arouinfar @kathy-phet @ariel-phet what do you recommend?

**Comment by ariel-phet** [comment]
@samreid there is definitely something odd here. I am not so disturbed by there being some ripples, but this behavior is not the same as Java.

What is actually more important in my opinion is that if you start off on "pulse" and then send a single pulse out, the entire background is eventually green. That seems pedagogically just wrong, it is like the space is just all filled with green light. Again, Java does not have this behavior. It is as if the lattice is not getting restored to black at all. 

Java also has the same general idea of mapping -- when the wave is running, "black" represents a trough, but when the wave is shut off, black represents zero. In Java, when the wave is shut off or a single pulse is given it seems to send a "make the lattice black wave" out behind the pulse. It would be nice to match this behavior....and it looks like the Java model has some specific logic to restore the lattice to black.

**Comment by ariel-phet** [comment]
@samreid when the pulse is used or the continuous wave is turned off, couldn't there be some wave sent out that is sort of invisible to the user that clears the lattice and puts in to the default starting condition? That seems to be what Java is doing?

**Comment by samreid** [comment]
Let's focus this issue on the the numerical artifacts in the lattice due to the discontinuity in the oscillator derivative.  I'll create a new issue about returning the light background to dark.

**Comment by samreid** [comment]
@ariel-phet I'll investigate the light view returning to black in #258.  Do you think we should investigate eliminating the discontinuity in the oscillator derivative that is leading to trailing ripples in the lattice values?

**Comment by samreid** [comment]
> @samreid please solve #258 first, and then we will revaluate this issue.

I'm happy to investigate #258, but wanted to make sure everyone is aware that #258 only applies to the light scene, and will not address artifacts seen on the water or sound scene.

**Comment by samreid** [comment]
@ariel-phet can we re-evaluate this issue before addressing #258 because #258 will not impact water or sound scenes at all?  Will work on this issue be essential before 1.0 publication?

**Comment by samreid** [comment]
I agree #258 is important and work on that in #258.

For this issue, the proposal from the preceding comment is: On screen 1, all scenes, when the source is shut off, to clear the play area.  I realize I proposed this in https://github.com/phetsims/wave-interference/issues/258#issuecomment-450871586, however, this feels like "sweeping the problem under the rug", and will lead to an inconsistency with "pulse" mode.  It seems problematic that running continuous for one cycle would look different than pulse.

> @samreid please solve #258 first, and then we will revaluate this issue.

Maybe once we have solved #258, we could leverage that solution to clear out the trailing ripples for sound and water. (Not setting the water/sound lattice to black, but setting the wave value to 0).  I'll put this back on hold and mention that as a possible avenue for exploration as part of #258.  However, I am not sure that is essential because @ariel-phet commented previously:

> I am not so disturbed by there being some ripples

So I'm confused about whether anything should be done for this issue at all or if the only problem is #258.  Until I hear further, I'll plan to continue in #258 keeping in mind we may wish to reuse that strategy to set wave values to zero for water and sound.

**Comment by samreid** [comment]
The proposed strategy in #258 is also handling this problem nicely:

![image](https://user-images.githubusercontent.com/679486/50739753-bffdd100-11a1-11e9-90aa-00f0dcba742f.png)

You can still see this effect a little bit if you shut off one source while another is still on, but the problem is significantly muted:

![image](https://user-images.githubusercontent.com/679486/50739787-14a14c00-11a2-11e9-9cb3-7bb2e1031ebe.png)

Come to think of it, we could even address that latter artifact if we introduce 2 new lattices.  (One per source and one for the sum of sources, then they could use the masking individually).  But this would be more complicated and would reduce performance.

**Comment by samreid** [comment]
On hold until I figure out #302

**Comment by arouinfar** [comment]
@samreid this appears to be addressed in dev.57, though dev.57 does have other quirks, as reported in https://github.com/phetsims/wave-interference/issues/304#issuecomment-452046234. I'm not sure if we should close this, since it may need to get re-tested once #304 is addressed. Leaving this one up to @samreid.

**Comment by samreid** [comment]
This seems much clearer after #300, closing.
